### PR TITLE
[chore] test shutdown behavior regardless of lifecycle skip flag

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
This changes how we test the shutdown of the components, making sure we test them regardless if they have been marked to skip because of the lifecycle.

**Link to tracking Issue:**
#17400 

**Testing:**
Only test changes.

**Documentation:**
N/A

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]
Should this have a changelog entry?  It is mostly a test change, but it can impact the way component developers build their components, as evidenced by the included changes to a number of receivers.

**Comment by atoulme** [comment]
Previous one didn't have one either: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17199

**Comment by atoulme** [comment]
Ah, I found a problem with this - this is added to the changes of this PR, make sure to take a look: the statsdreceiver component was starting its server in the `New` method instead of adhering to the start/shutdown cycle. I have moved the server creation to the start method.

File: receiver/statsdreceiver/receiver.go [hunk 1]
```diff
@@ -93,9 +87,14 @@ func buildTransportServer(config Config) (transport.Server, error) {
 // Start starts a UDP server that can process StatsD messages.
 func (r *statsdReceiver) Start(ctx context.Context, host component.Host) error {
 	ctx, r.cancel = context.WithCancel(ctx)
+	server, err := buildTransportServer(*r.config)
```
Comment (dmitryax) [inline receiver/statsdreceiver/receiver.go:90]: Is this change needed in the scope of this PR? If so let's move it to another PR since it can affect end users I believe

Comment (atoulme) [reply receiver/statsdreceiver/receiver.go:90]: It is needed as the tests reveal this bug. I can move it out to its own PR, sure!

Comment (atoulme) [reply receiver/statsdreceiver/receiver.go:90]: Moved to #17403, thanks!

**Comment by atoulme** [comment]
This PR now depends on #17403 , must be merged first to pass tests.

**Review by dmitryax — APPROVED** [review]
LGTM. Please rebase

**Comment by atoulme** [comment]
Done, thanks!

**Review by dmitryax — APPROVED** [review]
